tools/flaky-test-retryer/log_parser.go

```diff
@@ -21,22 +21,109 @@ package main
 
 import (
 	"fmt"
+	"log"
 	"path/filepath"
 	"strings"
 
 	"github.com/knative/test-infra/shared/junit"
 	"github.com/knative/test-infra/shared/prow"
 	"github.com/knative/test-infra/tools/flaky-test-reporter/jsonreport"
+	// TODO: remove this import once "k8s.io/test-infra" import problems are fixed
+	// https://github.com/test-infra/test-infra/issues/912
+	"github.com/knative/test-infra/tools/monitoring/prowapi"
 )
 
 // InitLogParser configures jsonreport's dependencies.
 func InitLogParser(serviceAccount string) error {
 	return jsonreport.Initialize(serviceAccount)
 }
 
+// JobData contains stripped-down information about a failed job, and local caches
+// of its failed tests and the flaky report it is referencing.
+type JobData struct {
+	Name         string
```

Comment:
I did do it this way initially, I changed it because when we use these fields later it is not too clear what is going on. Ex: `jd.Pull` vs `jd.Refs[0].Pulls[0].Number`. I can change it back if the other way is more expandable in the future.

Reply:
it will be slightly longer to write, but is more maintainable as we dont have to keep this struct in parity with prowapi and we can use the methods on the properties as well